### PR TITLE
Use the same version of Nokogiri as Vagrant

### LINKED ISSUE #768: Nokogiri upgraded in Vagrant 1.9.4
Installation of vagrant-libvirt via `vagrant plugin install vagrant-libvirt` and the longer Arch Linux version fail with the following error:

```
/opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/specification.rb:2100:in `check_version_conflict': can't activate nokogiri-1.6.8.1, already activated nokogiri-1.7.1 (Gem::LoadError)
	from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/specification.rb:1276:in `activate'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/bundler.rb:388:in `block in activate_solution'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/bundler.rb:385:in `each'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/bundler.rb:385:in `activate_solution'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/bundler.rb:288:in `internal_install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/bundler.rb:106:in `install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/plugin/manager.rb:62:in `block in install_plugin'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/plugin/manager.rb:72:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/plugin/manager.rb:72:in `install_plugin'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/commands/plugin/action/install_gem.rb:29:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/action/builder.rb:116:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/action/runner.rb:66:in `block in run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/action/runner.rb:66:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/commands/plugin/command/base.rb:14:in `action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/commands/plugin/command/install.rb:32:in `block in execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/commands/plugin/command/install.rb:31:in `each'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/commands/plugin/command/install.rb:31:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/plugins/commands/plugin/command/root.rb:66:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/cli.rb:42:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/lib/vagrant/environment.rb:308:in `cli'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.9.4/bin/vagrant:127:in `<main>'
```

It looks like Vagrant upgraded its [version of Nokogiri here](https://github.com/mitchellh/vagrant/blob/d790c138ac5c4755facb28fc688b8304209c961a/vagrant.gemspec#L36).

This is the longer recommended installation command for Arch Linux that doesn't make a difference.

```
CONFIGURE_ARGS='with-ldflags=-L/opt/vagrant/embedded/lib with-libvirt-include=/usr/include/libvirt with-libvirt-lib=/usr/lib' \                                    ⏎
  GEM_HOME=~/.vagrant.d/gems GEM_PATH=$GEM_HOME:/opt/vagrant/embedded/gems PATH=/opt/vagrant/embedded/bin:$PATH \
  vagrant plugin install vagrant-libvirt 
```

### PR DESCRIPTION
**Pull request by jcf**
Fixes #768.

Maybe… :smile:

```
$ vagrant --version
Vagrant 1.9.4
```

### CODE REVIEW

**Comment by jcf** [comment]
After attempting to install vagrant-libvirt with `sudo /opt/vagrant/embedded/bin/gem install vagrant-libvirt-0.0.37.gem` (which failed), running the lengthy Arch Linux installation command below worked!

```
$ CONFIGURE_ARGS='with-ldflags=-L/opt/vagrant/embedded/lib with-libvirt-include=/usr/include/libvirt with-libvirt-lib=/usr/lib' \
  GEM_HOME=~/.vagrant.d/gems GEM_PATH=$GEM_HOME:/opt/vagrant/embedded/gems PATH=/opt/vagrant/embedded/bin:$PATH \
  vagrant plugin install vagrant-libvirt 
Installing the 'vagrant-libvirt' plugin. This can take a few minutes...
Building native extensions.  This could take a while...
Installed the plugin 'vagrant-libvirt (0.0.37)'!
```

**Comment by pronix** [comment]
require resolve conflicts

**Comment by jcf** [comment]
@pronix rebased.

File: vagrant-libvirt.gemspec [hunk 1]
```diff
@@ -20,9 +20,11 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency "rspec-expectations", "~> 2.14.0"
   gem.add_development_dependency "rspec-mocks", "~> 2.14.0"
 
-  gem.add_runtime_dependency 'fog-libvirt', '>= 0.3.0'
-  gem.add_runtime_dependency 'nokogiri', '~> 1.6.0'
   gem.add_runtime_dependency 'fog-core', '~> 1.43.0'
+  gem.add_runtime_dependency 'fog-libvirt', '>= 0.3.0'
+
+  # Make sure to use the same version of Nokogiri as Vagrant (https://git.io/v9sSO).
+  gem.add_runtime_dependency 'nokogiri', '~> 1.7.1'
```
Comment (jcf) [inline vagrant-libvirt.gemspec]: There was a conflict here so I pulled the Nokogiri line down, and sorted the lines above in the hope that a bit of determinism will make future changes less likely to conflict.

**Comment by jcf** [comment]
The build's broken because of version conflicts:

```
Resolving dependencies...
Bundler could not find compatible versions for gem "nokogiri":
  In Gemfile:
    vagrant was resolved to 1.8.4, which depends on
      nokogiri (= 1.6.7.1)
    vagrant-libvirt was resolved to 0.0.40, which depends on
      nokogiri (~> 1.7.1)
```

**Comment by omahn** [comment]
`core: Remove nokogiri dependency and constraint [GH-8571]`

**Comment by ekohl** [comment]
That's https://github.com/mitchellh/vagrant/commit/bf1021905ba6a05d0ddf3df134e16f2b0c98f365 which is shipped since 1.9.5.

**Comment by electrofelix** [comment]
Well need to be able to switch the nokogiri dependency depending on which version of vagrant is installed, rather than just using multiple gemfiles for ci.

**Comment by agriffis** [comment]
Would this work, with old and new vagrants?

```ruby
gem.add_runtime_dependency 'nokogiri', '~> 1.6'
```

File: vagrant-libvirt.gemspec (hunk 1 above)
Comment (electrofelix) [inline vagrant-libvirt.gemspec]: Someone suggested using '~ 1.6' to allow this to remain working on older versions as this should then use which ever version is required by vagrant.

Comment (agriffis) [inline vagrant-libvirt.gemspec]: @electrofelix That was me. I think this should work:

```
gem.add_runtime_dependency 'nokogiri', '~> 1.6'
```

**Comment by hgfischer** [comment]
This bug is so annoying. Any simple workarounds?

**Comment by ikari7789** [comment]
I'm not a Ruby guru, but perhaps the way that the parallel's plugin handled it could be applied?

https://github.com/Parallels/vagrant-parallels/pull/298

**Comment by ikari7789** [comment]
@jcf Any chance you could update this to set the dependency to `~> 1.6` instead of `~> 1.7.1`?

File: vagrant-libvirt.gemspec (hunk 1 above)
Comment (kzorba) [reply vagrant-libvirt.gemspec]: @agriffis Your change works for me. Have a look at issue #828, I left a comment. Used vagrant 2.0.1 on Ubuntu 16.04 LTS. The only change needed seems to be
````
kzorba@enigma-home ~/tmp/vagrant-libvirt (master)-> git diff vagrant-libvirt.gemspec
diff --git a/vagrant-libvirt.gemspec b/vagrant-libvirt.gemspec
index f8fe7b5..a0b02b8 100644
--- a/vagrant-libvirt.gemspec
+++ b/vagrant-libvirt.gemspec
@@ -21,7 +21,7 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency "rspec-mocks", "~> 3.5.0"
 
   gem.add_runtime_dependency 'fog-libvirt', '>= 0.3.0'
-  gem.add_runtime_dependency 'nokogiri', '~> 1.6.0'
+  gem.add_runtime_dependency 'nokogiri', '~> 1.6'
   gem.add_runtime_dependency 'fog-core', '~> 1.43.0'
 
   gem.add_development_dependency 'rake'
````

I think @jcf PR should be modified slightly as mentioned. I must say however I am very new to vagrant and totally ignorant about Ruby ;-)

**Pull request merged**